### PR TITLE
Temporal cross validation on hyperparameter tuning

### PR DESCRIPTION
**Pull request by gmgivi**
In the previous implementation, temporal blocking is only applied on folds, but hyperparameters within the folds are tuned with GroupKFold, which ignores temporal autocorrelation and may select over‑optimistic parameters. This could be the reason that we didn't have much change in the R2 scores.

Code changes:
- In `optimize_hyperparameters`, when `use_temporal_cv=True`:
   - The cross‑validator passed to `cross_validate` becomes `HierarchicalTimeGroupCV(n_splits_tree=self.cv)`.
   - Groups are passed as a tuple `(tree_ids, period_idx, plot_id)` for `HierarchicalGroupCV`.
- The `fit` method forwards the same groups (and the `use_temporal_cv` flag) to `optimize_hyperparameters`

Results:

Analysis with ablation = 'all', model = 'gbdt', group = 'tree_id' with temporal grouping
 
| Species    | R² (test) mean ± std [all values]                     | RMSE   | n_test |
|------------|--------------------------------------------------------|--------|--------|
| spruce     | 0.21 ± 0.04 [0.16, 0.25, 0.19, 0.18, 0.25]            | 0.25   | 1025   |
| pine       | 0.19 ± 0.04 [0.13, 0.17, 0.22, 0.22, 0.22]            | 0.25   | 672    |
| beech      | 0.24 ± 0.04 [0.17, 0.23, 0.23, 0.28, 0.26]            | 0.24   | 529    |
| oak        | 0.02 ± 0.06 [-0.05, -0.02, 0.01, 0.11, 0.05]          | 0.27   | 199    |
| Weighted | 0.19                               | 0.25   | -      |
| Average | 0.16                                    | 0.25   | -      |

Old Results (using models.py from branch quantile_invert/main):
| Species    | R² (test) mean ± std [all values]                     | RMSE   | n_test |
|------------|--------------------------------------------------------|--------|--------|
| oak        | 0.30 ± 0.06 [0.20, 0.31, 0.32, 0.31, 0.37]            | 0.23   | 192    |
| pine       | 0.40 ± 0.05 [0.38, 0.40, 0.47, 0.33, 0.42]            | 0.22   | 655    |
| spruce     | 0.49 ± 0.03 [0.54, 0.50, 0.46, 0.47, 0.50]            | 0.20   | 1067   |
| beech      | 0.38 ± 0.02 [0.36, 0.37, 0.42, 0.39, 0.38]            | 0.22   | 528    |
| Weighted   | 0.43                                                  | 0.21   | -      |
| Average    | 0.39                                                  | 0.22   | -      |

### CODE REVIEW

**Comment by gmermoud** [comment]
@gmgivi temporal blocking must be applied for _performance estimation_. Hyperparameter optimization (HPO), as long as it is performed **only** on the `train` fold, is really up to us. The reason why performance collapses with using temporal blocking again here is because the second round of temporal blocking leaves too little training data and forces the HPO to regularize too aggressively. A regular tree-wise split is fair in this case, and doesn't cause leakage since we use temporal blocking of the test set.

Actually, I would push this further and use tree-wise splitting for HPO of plot-wise CV, just for the sake of completeness. *BUT*: this shows that we must be clearer in the paper about the difference between the CV that we use for performance evaluation, and the CV we use for HPO.

**Comment by gmgivi** [comment]
Decided to do hyperparameter optimization as in the previous implementation.
